src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -121,9 +119,8 @@ public String getZone() {
 
   @DataBoundSetter
   public void setZone(String zone) {
-    if (!Strings.isNullOrEmpty(zone)) {
-      this.zone = zone;
-    }
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(zone));
```

Comment:
I was able to confirm that when the UI only has an entry for `cluster` and not `zone` or `clusterName`, these setters are never called at all. Likewise, in the pipeline when only `zone` and `clusterName` are provided without `cluster`, that setter is not called. So, we can have these preconditions without any issues.

Reply:
Makes sense.